### PR TITLE
Refactor OCPControllerManager to conform to the internal service manager

### PR DESCRIPTION
**Pull request by husky-parul**
Refactor OCPControllerManager to conform to the internal service manager #239

Signed-off-by: Parul <parsingh@redhat.com>

**Which issue(s) this PR addresses**:

Closes#239

### CODE REVIEW

File: pkg/util/net.go [hunk 1]
```diff
@@ -52,3 +53,21 @@ func RetryInsecureHttpsGet(url string) int {
 
 	return status
 }
+
+func RetryTCPConnection(host string, port string) int {
+	status := 0
+	err := wait.Poll(5*time.Second, 120*time.Second, func() (bool, error) {
+		timeout := 3 * time.Second
+		_, err := tcpnet.DialTimeout("tcp", tcpnet.JoinHostPort("127.0.0.1", "8445"), timeout)
```
Comment (rootfs) [inline pkg/util/net.go:61]: host and port

File: pkg/util/net.go [hunk 2]
```diff
@@ -52,3 +53,21 @@ func RetryInsecureHttpsGet(url string) int {
...
+
+		if err == nil {
+			status = 200
```
Comment (rootfs) [inline pkg/util/net.go:64]: 200 is HTTP status. If this is for TCP port probe, returning true/false is good to go

File: pkg/util/net.go [hunk 3]
```diff
@@ -52,3 +53,21 @@ func RetryInsecureHttpsGet(url string) int {
 
 	return status
 }
+
+func RetryTCPConnection(host string, port string) int {
+	status := 0
+	err := wait.Poll(5*time.Second, 120*time.Second, func() (bool, error) {
+		timeout := 3 * time.Second
```
Comment (rootfs) [inline pkg/util/net.go:60]: can you check what's the default timeout in `net/http`? 3 seconds could be too short

Comment (husky-parul) [reply pkg/util/net.go:60]: The default timeout is no timeout so it will keep trying infinitely. 
https://github.com/golang/go/issues/24138

Increasing it to `30 * time.Second`, I see it is the recommended timeout -> https://pkg.go.dev/net/http

File: hack/all-in-one/README.md [hunk 4]
```diff
@@ -68,17 +68,17 @@ kubectl get pods -A
 ### Access the cluster on the host
 #### Linux
 ```bash
-export KUBECONFIG=$(podman volume inspect ushift-vol --format "{{.Mountpoint}}")/microshift/resources/kubeadmin/kubeconfig
+export KUBECONFIG=$(podman volume inspect microshift-data --format "{{.Mountpoint}}")/microshift/resources/kubeadmin/kubeconfig
```
Comment (rootfs) [inline hack/all-in-one/README.md:71]: can you revert these changes?

File: pkg/util/net.go [hunk 5]
```diff
@@ -52,3 +53,21 @@ func RetryInsecureHttpsGet(url string) int {
...
 }
+
+func RetryTCPConnection(host string, port string) bool {
+	status := true
+	err := wait.Poll(5*time.Second, 120*time.Second, func() (bool, error) {
+		timeout := 30 * time.Second
+		_, err := tcpnet.DialTimeout("tcp", tcpnet.JoinHostPort(host, port), timeout)
+
+		if err == nil {
+			status = false
```
Comment (rootfs) [inline pkg/util/net.go:64]: why set it false?

File: hack/all-in-one/README.md (hunk 4 above)
Comment (husky-parul) [reply hack/all-in-one/README.md:71]: I will. Can't get the tests to pass though. :thinking:

File: pkg/util/net.go [hunk 6]
```diff
@@ -52,3 +53,21 @@ func RetryInsecureHttpsGet(url string) int {
...
+			return true, nil
+		}
+		return false, nil
+	})
+	if err != nil && err == wait.ErrWaitTimeout {
+		logrus.Warningf("Endpoint is not returning any status code")
```
Comment (rootfs) [inline pkg/util/net.go:70]: if timeout, this should return false to tell the caller the connection is not ready

File: hack/all-in-one/README.md (hunk 4 above)
Comment (husky-parul) [reply hack/all-in-one/README.md:71]: > can you revert these changes?

Will the changes to volume name will go in a different PR or we don't need this change at all? All other places use `microshift-data` as the volume name instead of `ushift-vol`

**Comment by rootfs** [comment]
@oglok @mangelajo please take a final review
